spec/users_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative './spec_helper'
+
+module ThreeScale
+  module Core
+    describe Users do
+      let(:service_id) { 2001 }
+      let(:provider_key) { 'provider_for_user_spec' }
+      let(:state) { :active }
+
+      before do
+        Service.save! provider_key: provider_key, id: service_id, default_service: true
+      end
+
+      describe '.delete!' do
+        it 'deletes all users' do
+          Users.delete!(service_id).must_equal true
```

Comment:
With this, we can't really be sure about whether the users were deleted.

I would create some and then verify that they are no longer there. We have methods for that in the `User` class.

Reply:
I will implement. But I consider tests in pisoni should not check about functionality and logic from apisonator.

Reply:
In apisonator there should be tests to check that https://github.com/3scale/apisonator/pull/38